js/src/Components/Widgets/LinkWidget.js

```diff
@@ -118,48 +142,64 @@ export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) =>
   /**
    * A suggestion was picked, pass that up to the manager to store the value.
    */
-  const suggestionPicked = (e, selectedValue) => {
-    onFieldChange([{
-      title: defaultFieldValue.title,
-      uri: getUriFromString(selectedValue === null ? '' : selectedValue.value)
-    }]);
+  const suggestionPicked = (e, selectedValue, delta) => {
+    alterValues({
+      title: fieldValues[delta].title,
+      uri: getUriFromString(selectedValue === null ? '' : selectedValue.value),
+      delta: delta
+    });
   };
 
   /**
    * When the textfield on the URI blurs, that's when we want to trigger the field change and pass it up to the manager.
    */
-  const onUriBlur = (e) => {
-    onFieldChange([{
-      title: defaultFieldValue.title,
-      uri: getUriFromString(e.target.value)
-    }]);
+  const onUriBlur = (e, delta) => {
+    alterValues({
+      title: fieldValues[delta].title,
+      uri: getUriFromString(e.target.value),
+      delta: delta
+    });
   }
 
-  return (
-    <FormGroup>
-      <FormLabel component="legend">
-        {settings.label}
-      </FormLabel>
+  /**
+   * To support cardinality, we will need an "Add another" button.
+   */
+  const addAnotherButton = (cardinality) => {
```

Comment:
theres no need for passing `cardinality` here. the function will already have access to `settings.cardinality`

Reply:
fixed